src/main/scala/com/fulcrumgenomics/personal/nhomer/GenerateRegionsFromFasta.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.personal.nhomer
+
+import com.fulcrumgenomics.cmdline.{Personal, JeanLucTool}
+import dagr.commons.CommonsDef.PathToFasta
+import dagr.commons.io.Io
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools.SAMSequenceDictionary
+import htsjdk.samtools.reference.{ReferenceSequenceFile, ReferenceSequenceFileFactory}
+
+import java.nio.file.Path
+import scala.collection.JavaConversions._
+
+@clp(
+  description="Generates a list of freebayes/bamtools region specifiers.",
+  group = classOf[Personal]
+)
+class GenerateRegionsFromFasta
+( @arg(doc = "The input FASTA.") val input: PathToFasta,
+  @arg(doc = "The output.") val output: Path = Io.StdOut,
+  @arg(doc = "The size of the regions to output.") val regionSize: Int = 100000
+) extends JeanLucTool with LazyLogging {
+
+  Io.assertReadable(input)
```

Comment:
I did meant to put them there.  Perhaps you are thinking to much in `dagr` where `input` could be not readable at construction time, but here I am putting it in the constructor since it is a simple tool.  I can easily put it into `execute` if you ever think you call this class directly.